sway-core/src/language/ty/declaration/variable.rs

```diff
@@ -7,6 +7,7 @@ pub struct TyVariableDeclaration {
     pub name: Ident,
     pub body: TyExpression,
     pub mutability: VariableMutability,
+    pub return_type: TypeId,
     pub type_ascription: TypeId,
```

Comment:
I'm actually questionning the usefulness of `type_ascription` here 🤔  - we don't seem to have it for `ConstantDeclaration` and it's been consumed already. All we care about is the actual type now which `return_type` will hold. cc @emilyaherbert what do you think?